cubecl-book/src/getting-started/benchmark.md

```diff
@@ -0,0 +1,246 @@
+# Benchmarking reduction
+Now that we have a basic understanding of how to perform a reduction operation, let's benchmark it to see how it performs in terms of speed and efficiency.
+
+## Benchmarking struct
+For benchmarking, we will create a struct that holds the necessary information for the benchmark, such as the input shape, device, and client. This struct will be used to run the benchmark tests and configure the benchmarking environment. Please note that the `Runtime` and `Float` traits are used to make the benchmark generic over different CubeCL runtimes and floating-point types. A `PhantomData` is used to indicate that the struct holds a type parameter `F` without actually storing a value of that type, which is useful for generic programming in Rust for more information see the [Rust documentation](https://doc.rust-lang.org/std/marker/struct.PhantomData.html).
+```rust,ignore
+# use std::marker::PhantomData;
+#
+# use cubecl::benchmark::{Benchmark, TimingMethod};
+# use cubecl::server::Handle;
+# use cubecl::std::tensor::compact_strides;
+# use cubecl::{future, prelude::*};
+
+pub struct ReductionBench<R: Runtime, F: Float> {
+    input_shape: [usize; 2],
+    device: R::Device,
+    client: ComputeClient<R::Server, R::Channel>,
+    _phantom_data: PhantomData<F>,
+}
+```
+## Implementing the benchmark trait
+To benchmark a CubeCL kernel, it is recommended to implement the `Benchmark` trait that defines the necessary methods for preparing, executing, and synchronizing the benchmark, because GPU are asynchronous and most benchmarking tool will not wait for the GPU to finish executing the kernel before measuring the time it took to execute it with a sync.
+```rust,ignore
+pub trait Benchmark {
+    /// Benchmark arguments.
+    type Args: Clone;
```

Comment:
On main it is now `Input` and `Output`. I would update that code for use the latest version.

Reply:
done